# Profitable App Profiles for the App Store and Google Play Markets
## By Jorge Hernandez Leon
### This project is to find mobile app profiles that are profitable for the App Store and Google Play markets. Our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.
### My goal for this project is to analyze data to help you developers, understand what kinds of apps are likely to attract more users.

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
#android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
#ios = ios[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
def freq_table(dataset, index_freq):
    freq_table1 = {}
    for n in dataset:
        freq_counter = n[index_freq]
        if freq_counter in freq_table1:
            freq_table1[freq_counter] += 1
        else:
            freq_table1[freq_counter] = 1
    #for variable in freq_table1: # inpercentage form:
     #   freq_table1[variable] /= len(dataset)
      #  freq_table1[variable] *= 100
    return freq_table1

In [4]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [5]:
number1 = 0
for n in android[0]:
    print(n, "index is", number1)
    number1 += 1

App index is 0
Category index is 1
Rating index is 2
Reviews index is 3
Size index is 4
Installs index is 5
Type index is 6
Price index is 7
Content Rating index is 8
Genres index is 9
Last Updated index is 10
Current Ver index is 11
Android Ver index is 12


In [6]:
number1 = 0
for n in ios[0]:
    print(n, "index is", number1)
    number1 += 1

 index is 0
id index is 1
track_name index is 2
size_bytes index is 3
currency index is 4
price index is 5
rating_count_tot index is 6
rating_count_ver index is 7
user_rating index is 8
user_rating_ver index is 9
ver index is 10
cont_rating index is 11
prime_genre index is 12
sup_devices.num index is 13
ipadSc_urls.num index is 14
lang.num index is 15
vpp_lic index is 16


# We explore both Apple and Android apps to see what type of app could possibly attract to more user across consumers interface. 

In [7]:
explore_data(ios,0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


Number of rows: 7198
Number of columns: 17


# Cleaning Data

In [8]:
freq_table(android, 7)
# one of the price is called everyone

{'Price': 1,
 '0': 10040,
 '$4.99': 72,
 '$3.99': 63,
 '$6.99': 13,
 '$1.49': 46,
 '$2.99': 129,
 '$7.99': 7,
 '$5.99': 30,
 '$3.49': 7,
 '$1.99': 73,
 '$9.99': 21,
 '$7.49': 2,
 '$0.99': 148,
 '$9.00': 2,
 '$5.49': 6,
 '$10.00': 3,
 '$24.99': 7,
 '$11.99': 5,
 '$79.99': 2,
 '$16.99': 3,
 '$14.99': 11,
 '$1.00': 3,
 '$29.99': 7,
 '$12.99': 5,
 '$2.49': 26,
 '$10.99': 2,
 '$1.50': 1,
 '$19.99': 6,
 '$15.99': 1,
 '$33.99': 2,
 '$74.99': 1,
 '$39.99': 2,
 '$3.95': 2,
 '$4.49': 9,
 '$1.70': 2,
 '$8.99': 5,
 '$2.00': 3,
 '$3.88': 1,
 '$25.99': 1,
 '$399.99': 12,
 '$17.99': 2,
 '$400.00': 1,
 '$3.02': 1,
 '$1.76': 1,
 '$4.84': 1,
 '$4.77': 1,
 '$1.61': 1,
 '$2.50': 1,
 '$1.59': 1,
 '$6.49': 5,
 '$1.29': 1,
 '$5.00': 1,
 '$13.99': 2,
 '$299.99': 1,
 '$379.99': 1,
 '$37.99': 1,
 '$18.99': 1,
 '$389.99': 1,
 '$19.90': 1,
 '$8.49': 2,
 '$1.75': 1,
 '$14.00': 1,
 '$4.85': 1,
 '$46.99': 1,
 '$109.99': 1,
 '$154.99': 1,
 '$3.08': 1,
 '$2.59': 1,
 '$4.80': 1,
 '$1.96': 1,
 '$19.40': 1,
 '$3.90': 1,


In [9]:
#Removing Everyone
row = 0

for n in android[0:]:
    if str('Everyone') == n[7]:
        print(row)
    row += 1
    
    
    

10473


In [10]:
#error is on row 10473 on android(header)
explore_data(android,10473, 10474, True)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


Number of rows: 10842
Number of columns: 13


In [11]:
 del android[10473]

In [12]:
#with deleted bad data
explore_data(android, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


### Remove duplicates

In [13]:
freq_table(android,0)

{'App': 1,
 'Photo Editor & Candy Camera & Grid & ScrapBook': 1,
 'Coloring book moana': 2,
 'U Launcher Lite – FREE Live Cool Themes, Hide Apps': 1,
 'Sketch - Draw & Paint': 1,
 'Pixel Draw - Number Art Coloring Book': 1,
 'Paper flowers instructions': 1,
 'Smoke Effect Photo Maker - Smoke Editor': 1,
 'Infinite Painter': 1,
 'Garden Coloring Book': 1,
 'Kids Paint Free - Drawing Fun': 1,
 'Text on Photo - Fonteee': 1,
 'Name Art Photo Editor - Focus n Filters': 1,
 'Tattoo Name On My Photo Editor': 1,
 'Mandala Coloring Book': 1,
 '3D Color Pixel by Number - Sandbox Art Coloring': 1,
 'Learn To Draw Kawaii Characters': 1,
 'Photo Designer - Write your name with shapes': 1,
 '350 Diy Room Decor Ideas': 1,
 'FlipaClip - Cartoon animation': 1,
 'ibis Paint X': 1,
 'Logo Maker - Small Business': 1,
 "Boys Photo Editor - Six Pack & Men's Suit": 1,
 'Superheroes Wallpapers | 4K Backgrounds': 1,
 'Mcqueen Coloring pages': 2,
 'HD Mickey Minnie Wallpapers': 1,
 'Harley Quinn wallpapers HD':

In [14]:
#finding duplicates
duplicate_apps = []
unique_apps = []

for app in android[1:]:
    name = app[0]
    if name in unique_apps:#if duplicate then add to redo list
        duplicate_apps.append(name)
    else: 
        unique_apps.append(name)
    
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


There are duplicate rows in android list. We will delete the rows with the dupicate, but keep the ones with the higher rating count. 

In [15]:
#Dictionary with unique apps and the highest review count
reviews_max = {}

for n in android[1:]:
    name = n[0]
    n_reviews = float(n[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max.update({name:n_reviews})
        # alternate script reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max.update({name:n_reviews})

print("Expected length:", len(android[1:])-len(duplicate_apps))
print("Actual length:", len(reviews_max))

Expected length: 9659
Actual length: 9659


Remove Duplicates by looking for the the name ff the app in the dictionary and if it has the same number of reviews as the dictionary. We will keep the entire row and add to the new list. Also the name will be added to the list already added to exclude duplicate row entries. 

In [16]:
#remove duplicates

android_clean = []
already_added = []

for row in android[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if name not in already_added and reviews_max[name] == n_reviews:
        android_clean.append(row)
        # alternate script reviews_max[name] = n_reviews
        already_added.append(name)
explore_data(android_clean, 0, 5, True)
    

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


In [17]:
#finding duplicates in ios
duplicate_apps_ios = []
unique_apps_ios = []

for app in ios[1:]:
    name = app[1]
    if name in unique_apps:#if duplicate then add to redo list
        duplicate_apps_ios.append(name)
    else: 
        unique_apps_ios.append(name)
    
print('Number of duplicate apps ios:', len(duplicate_apps_ios))
print('\n')
print('Examples of duplicate apps ios:', duplicate_apps_ios[:3])

Number of duplicate apps ios: 0


Examples of duplicate apps ios: []


Remove Non-English apps in ios lists

In [18]:
def split(word):
    return [char for char in word]
def unknown_string(string):
    list_string = split(string)
    count1 = 0
    for n in list_string:
        if ord(n) > 127:
            count1 += 1
        if count1 > 3:
            return False
    return True

In [19]:
unknown_string('Docs To Go™ Free Office Suite')

True

# Use the new function to filter out Non-English apps
### Will use android_clean and ios

In [20]:
android_clean_eng = []
android_clean_non_eng = []
for n in android_clean:
    title = n[0]
    if unknown_string(str(title)):
        android_clean_eng.append(n)
    elif not unknown_string(str(title)):
        android_clean_non_eng.append(n)
explore_data(android_clean_eng, 6150, 6155, True)
#explore_data(android_clean_non_eng, 0, 5, True)

['Ultimate Public Campgrounds (Over 37,100 in US&CA)', 'TRAVEL_AND_LOCAL', '4.7', '213', '6.1M', '5,000+', 'Paid', '$3.99', 'Everyone', 'Travel & Local', 'August 5, 2018', '1.8.8.0', '4.4 and up']


['20 minutes (CH)', 'NEWS_AND_MAGAZINES', '3.7', '4379', 'Varies with device', '1,000,000+', 'Free', '0', 'Teen', 'News & Magazines', 'August 3, 2018', 'Varies with device', 'Varies with device']


['OurHome – chores, rewards, groceries and calendar', 'FAMILY', '4.3', '3146', 'Varies with device', '100,000+', 'Free', '0', 'Everyone', 'Lifestyle;Education', 'January 31, 2018', '3.13.2', '4.1 and up']


['Weather Data CH', 'WEATHER', 'NaN', '15', 'Varies with device', '500+', 'Paid', '$2.99', 'Everyone', 'Weather', 'August 9, 2016', 'Varies with device', 'Varies with device']


['Dots puzzle', 'FAMILY', '4.0', '179', '14M', '50,000+', 'Paid', '$0.99', 'Everyone', 'Puzzle', 'April 18, 2018', 'NaN', '4.0 and up']


Number of rows: 9614
Number of columns: 13


In [21]:
explore_data(ios, 0 , 5, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


Number of rows: 7198
Number of columns: 17


In [22]:
ios_clean_eng = []
ios_clean_non_eng = []
for n in ios[1:]:
    title = n[2]
    if unknown_string(str(title)):
        ios_clean_eng.append(n)
    elif not unknown_string(str(title)):
        ios_clean_non_eng.append(n)
explore_data(ios_clean_eng, 0, 5, True)
#explore_data(ios_clean_non_eng, 0, 5, True)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


Number of rows: 6183
Number of columns: 17


# Isolate free variables

In [23]:
# first have to check how the prices come up
#ios
prices = {}
for n in ios[1:]:
    price = n[5]
    prices[price]=1 

print(prices)

{'3.99': 1, '0': 1, '0.99': 1, '9.99': 1, '4.99': 1, '7.99': 1, '2.99': 1, '1.99': 1, '5.99': 1, '12.99': 1, '21.99': 1, '249.99': 1, '6.99': 1, '74.99': 1, '19.99': 1, '8.99': 1, '24.99': 1, '13.99': 1, '14.99': 1, '16.99': 1, '47.99': 1, '11.99': 1, '59.99': 1, '15.99': 1, '27.99': 1, '17.99': 1, '299.99': 1, '49.99': 1, '23.99': 1, '20.99': 1, '39.99': 1, '99.99': 1, '29.99': 1, '34.99': 1, '18.99': 1, '22.99': 1}


In [24]:
# first have to check how the prices come up
#android clean eng
prices= {}
for n in android_clean_eng:
    price = n[6]
    prices[price] = 1 
    if price == 'NaN':
        print(n)

print(prices)

['Command & Conquer: Rivals', 'FAMILY', 'NaN', '0', 'Varies with device', '0', 'NaN', '0', 'Everyone 10+', 'Strategy', 'June 28, 2018', 'Varies with device', 'Varies with device']
{'Free': 1, 'Paid': 1, 'NaN': 1}


In [25]:
free_ios_apps = []
for n in ios_clean_eng:
    price = float(n[5])
    if price == 0:
        free_ios_apps.append(n)
explore_data(free_ios_apps,0,5, True)

['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['7', '283646709', 'PayPal - Send and request money safely', '227795968', 'USD', '0', '119487', '879', '4', '4.5', '6.12.0', '4+', 'Finance', '37', '0', '19', '1']


Number of rows: 3222
Number of columns: 17


In [26]:
free_android_apps = []
for n in android_clean_eng:
    price = n[7]
    if price == '0':
        free_android_apps.append(n)
print(len(free_android_apps))

8864


## After cleaning the data we are now ready to find connections between the two datasets. We want to find a correlation between the two datasets in order to find an app that is like by both interface users. 

In [27]:
explore_data(free_android_apps,0,100, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+

In [28]:
explore_data(free_ios_apps, 0, 3 , True)

['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


Number of rows: 3222
Number of columns: 17


### For the ios apps I will look at the index value, index = -5, the prime_genre column. Accordinaly in the android dataset we will investigate the column with the index 1 for category, and index 9 for genre.

In [29]:
# Making a Dictionary of a certain column results in Percentage
def freq_table(dataset, index_freq):
    freq_table1 = {}
    for n in dataset:
        freq_counter = n[index_freq]
        if freq_counter in freq_table1:
            freq_table1[freq_counter] += 1
        else:
            freq_table1[freq_counter] = 1
    for variable in freq_table1: # in percentage form:
        freq_table1[variable] /= len(dataset)
        freq_table1[variable] *= 100
    return freq_table1

In [30]:
#The sorted() function doesn't work too well with dictionaries because it only considers and returns the dictionary keys.
# So we have to do this to sorted out
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    

In [31]:
display_table(free_ios_apps, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


### The most popular app in ios is games with a 58% and then second with a big difference in percentage is Entertainment with 7.8% and third is Photo with a 5%. There are only 23 different genres in the ios system.

Most apps in are for entertainment(games, photo & video, social networking, sports and music) for a 70.6%. Practical apps account for 12.1 %. This is just a percentage of apps available for free in the ios system. This does not imply you should go with entertainment apps. You will be competeing with a lot of other apps if your app is an entertainment app. 

In [32]:
display_table(free_android_apps,1) # Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [33]:
display_table(free_android_apps,9) # Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The most common category is family, games and tools with their percentage respectively being 18.9%,9.7% and 8.5%. 
The genre column also shows practical apps being popular with Tools, Enterntainment and education being the top three with 8.4%, 6.1% and 5.3% being their percentages respectively. 

In [40]:
#Average number of user ratings 
genres_ios = freq_table(free_ios_apps, -5)#prime genre
table3 = {}
for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in free_ios_apps:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[6])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    table3[genre] = avg_n_ratings
#organize the last list
table_display = []
for key in table3:
    key_val_as_tuple = (table3[key], key)
    table_display.append(key_val_as_tuple)
        
table_sorted = sorted(table_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews.

In [48]:
#Average number of intalls per app genre
genres_android = freq_table(free_android_apps, 1)#Category
table4 = {}
for genre in genres_android:
    total = 0
    len_category = 0
    for app in free_android_apps:
        genre_app = app[1]
        if genre_app == genre:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    table4[genre] = avg_n_installs
    
#organize the last list
table_display = []
for key in table4:
    key_val_as_tuple = (table4[key], key)
    table_display.append(key_val_as_tuple)
        
table_sorted = sorted(table_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs: